tests/math/test_functions.py

```diff
@@ -2266,3 +2264,138 @@ def test_sort(self, input, test_output):
 
 
 ones_functions = [onp.ones, np.ones, jnp.ones, torch.ones, tf.ones]
+
+# fmt: off
+state_vectors = [
+    ([1, 0, 0, 0], ([[1, 0], [0, 0]], [[1, 0], [0, 0]], [[1, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0]])),

```

Comment:
This section is challenging to read.

Readability could be improved by:
1. Creating the input - expected pairs only once as lists and then using the created objects later for the different data structures;
2. Creating the matrices via NumPy (would provide a succinct def in exchange of a visual repr of the matrices):
```py
mat = np.zeros(4,4)
mat[0,0] = 1
```
3. Adding separator comments in between lines, e.g., `# List` and `# Vanilla NumPy Array`.

E.g.,
```python
00_state_vector = [1, 0, 0, 0]
00_wire_1_mx = [[1, 0], [0, 0]]
00_wire_2_mx = [[1, 0], [0, 0]]
00_full_mx = [[1, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0]]
```